### PR TITLE
button-behavior test V0

### PR DESCRIPTION
**Pull request by patole**
This test has just a square button in the middle of the scene and text above it. The text changes from the default state to indicate if the button detects either a quick click or a hold, then back to the default text after 1000 ms.

### CODE REVIEW

File: packages/functional-tests/src/tests/button-behavior-test.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+					height: .1,
+					anchor: MRE.TextAnchorLocation.BottomCenter,
+					color: MRE.Color3.Teal()
+				}
+			}
+		})
+
+		const testButtonBehavior = this.testButton.setBehavior(MRE.ButtonBehavior);
+		testButtonBehavior.onClick((_, __) => this.displayString("Click Detected!"));
+		testButtonBehavior.onButton('holding', () => this.displayString("Hold Detected!"));
```
Comment (tombuMS) [inline packages/functional-tests/src/tests/button-behavior-test.ts:88]: >'holding' [](start = 30, length = 9)

I see this is written as v0 in the title.  Is the desire to expand this to cover and test the other events for the button?

Comment (patole) [reply packages/functional-tests/src/tests/button-behavior-test.ts:88]: Good question! I wrote this based on a sentence or two description from @sorenhan and wanted his input to see if there were more events desired in this, or if simply a click/press threshold test was sufficient.

Comment (tombuMS) [reply packages/functional-tests/src/tests/button-behavior-test.ts:88]: Well as the behavior system area owner, I would totally love to expand this and fully test the button.  Let me know when you have time to discuss expanding this if you wanted to take that on as well.  :)

**Review by tombuMS — APPROVED** [review]
:shipit:

File: packages/functional-tests/src/tests/button-behavior-test.ts (hunk 1 above)
Comment (patole) [reply packages/functional-tests/src/tests/button-behavior-test.ts:88]: Sure thing, we can write this up after you're done with the media player work so as not to distract from that.

**Pull request merged**